servlet/src/main/java/com/linecorp/armeria/server/servlet/ServletHttpExchange.java

```diff
@@ -0,0 +1,160 @@
+        final DefaultFullHttpRequest fullHttpRequest = parseHttpRequest(context, request);
+        final ServletHttpExchange instance = new ServletHttpExchange();
+        instance.servletContext = servletContext;
+        instance.channelHandlerContext = context;
+
+        //Create a new servlet request object
+        instance.request = ServletHttpRequest.newInstance(instance, fullHttpRequest);
+        //Create a new servlet response object
+        instance.response = ServletHttpResponse.newInstance(instance);
+        return instance;
+    }
+
+    /**
+     * Whether the pipe is active.
+     */
+    public static boolean isChannelActive(DefaultServiceRequestContext channelHandlerContext) {
+        return true;
+    }
+
+    /**
+     * Is http keep alive.
+     */
+    public boolean isHttpKeepAlive() {
+        return isHttpKeepAlive;
+    }
+
+    /**
+     * Get http servlet request.
+     */
+    public ServletHttpRequest getRequest() {
+        return request;
+    }
+
+    /**
+     * Get servlet context.
+     */
+    public ServletContextImpl getServletContext() {
+        return servletContext;
+    }
+
+    /**
+     * Get http servlet response.
+     */
+    public ServletHttpResponse getResponse() {
+        return response;
+    }
+
+    /**
+     * Get channel handler context.
+     */
+    public DefaultServiceRequestContext getChannelHandlerContext() {
+        return channelHandlerContext;
+    }
+
+    /**
+     * Get server address.
+     */
+    public InetSocketAddress getServerAddress() {
+        return servletContext.getServerAddress();
+    }
+
+    /**
+     * Get local address.
+     */
+    public InetSocketAddress getLocalAddress() {
+        final SocketAddress socketAddress = channelHandlerContext.localAddress();
+        if (socketAddress == null) {
+            return null;
+        }
+        if (socketAddress instanceof InetSocketAddress) {
+            return (InetSocketAddress) socketAddress;
+        }
+        return null;
+    }
+
+    /**
+     * Get remote address.
+     */
+    public InetSocketAddress getRemoteAddress() {
+        final SocketAddress socketAddress = channelHandlerContext.remoteAddress();
+        if (socketAddress == null) {
+            return null;
+        }
+        if (socketAddress instanceof InetSocketAddress) {
+            return (InetSocketAddress) socketAddress;
+        }
+        return null;
+    }
+
+    /**
+     * Convert {@link HttpRequest} to {@link DefaultFullHttpRequest}.
+     */
+    private static DefaultFullHttpRequest parseHttpRequest(ServiceRequestContext ctx,
+                                                          AggregatedHttpRequest request) {
+        final ByteBuf content = Unpooled.wrappedBuffer(request.content().array());
+        final DefaultFullHttpRequest fullHttpRequest = new DefaultFullHttpRequest(
+                HttpVersion.HTTP_1_1, new HttpMethod(request.method().name()),
```

Comment:
Is it always use `HTTP/1.1`?

Reply:
No, but FullHttpRequest interface use HttpVersion which is not support all HTTP version, and we don't use this HttpVersion information. I set correct HTTP version from Armeria to ServletAPI now.